pilot/pkg/config/kube/crd/client.go

```diff
@@ -349,11 +349,20 @@ func (cl *Client) Get(typ, name, namespace string) (*model.Config, bool) {
 	}
 
 	config := s.object.DeepCopyObject().(IstioObject)
-	err := rc.dynamic.Get().
-		Namespace(namespace).
-		Resource(ResourceName(schema.Plural)).
-		Name(name).
-		Do().Into(config)
+	var err error
+
+	if namespace == "" {
+		err = rc.dynamic.Get().
+			Resource(ResourceName(schema.Plural)).
+			Name(name).
+			Do().Into(config)
+	} else {
+		err = rc.dynamic.Get().
+			Namespace(namespace).
+			Resource(ResourceName(schema.Plural)).
+			Name(name).
+			Do().Into(config)
+	}
```

Comment:
The request could also be built incrementally leveraging the builder pattern mechanics, i.e.

```golang
req := rc.dynamic.Get().Resource(ResourceName(schema.Plural)).Name(name)
if namespace != "" {
    req = req.Namespace(namespace)
}
err = req.Do().Into(config)
```